src/liballoc/collections/btree/map.rs

```diff
@@ -2760,8 +2795,8 @@ impl<'a, K: 'a, V: 'a> Handle<NodeRef<marker::Mut<'a>, K, V, marker::LeafOrInter
 impl<K, V> node::Root<K, V> {
     /// Removes empty levels on the top, but keep an empty leaf if the entire tree is empty.
     fn fix_top(&mut self) {
-        while self.height() > 0 && self.as_ref().len() == 0 {
-            self.pop_level();
+        while self.pop_empty_level() {
+            // Is it possible the child would be empty too? There is no test demonstrating that.
```

Comment:
Hm, I am not sure I entirely follow this comment. It seems like if the child is also empty, we'd just hit the next loop -- but indeed I think that we should never be in a case where two levels have empty nodes in the tree. I could be wrong (I'd have to re-read some of the docs and code).

Reply:
Actually, I shouldn't be so lazy... it's only for split_off, so let's write a test that splits off just the last leaf child of a 3 level tree... assuming there isn't one.

It's definitely impossible after a single remove, because that won't even condone underful children.

Reply:
It is possible to have multiple empty levels in the course of split_off. Whether you need that fix_top both at the beginning and at the end of fix_left_border/fix_right_border is something I'll leave in the middle. Now let's just hope that Miri doesn't find more aliasing errors with these test cases...